docs/ops/matrix/Einsum_7.md

```diff
@@ -0,0 +1,248 @@
+`<subscript for input1>, <subscript for input2>, ..., <subscript for inputn> -> <subscript for output>`.
+Each input subscript `<subscript for input1>` contains a sequence of labels (lower case letters `['a',...,'z']`) where each label refers to a dimension of 
+the corresponsing operand. The appereance of labels in a subscript in alphabetical order is not needed. The subscript for a scalar input is empty.
+The input subscripts are separated with a comma `,`.
+The output subscript `<subscript for output>` is separated from the input subscripts by `->` and also represents a sequence of labels
+(lower case letters `['a',...,'z']`). A length of the input subscript matches a rank of the input. The input subscript is empty for a scalar input.
+
+*Einsum* operation on multiple inputs can be treated as several consecutive *Einsum* operations. In the first step, *Einsum* applies the first two inputs. 
+In the second step, it operates on the result of the first step and the third input and so forth.
+*Einsum* operates on two operands similar to element-wise multiplication by all pairs of batches from both operands. The batch dimensions are defined
+with labels that enter only one subscript from the corresponding two subscripts.
+For example, the intermediate result after the first step for *Einsum* with three inputs (of shapes `[2, 5]`, `[5, 3, 6]` and `[5, 3]`)
+and `equation` equal to `ab,bcd,bc->ca` will be a tensor of shape `[2, 5, 3, 6]` with a subscript `abcd`
+where batch dimensions for the first input and the second input are represented with label sequences `a` and `cd`.
+The next step performs the same logic on input tensors (of shapes `[2, 5, 3, 6]` and `[5, 3]`) with subscripts `abcd` and `bc` and
+outputs a tensor of shape `[2, 5, 3, 6]` with a subscript `abcd`.
+Lastly, the output subscript defines order of output dimensions and which dimensions are sum-reduced.
+Dimensions corresponding to absent labels in the output subscript are sum-reduced. The final result for the considered example is of shape equal to `[3,2]`
+where dimensions with labels `b` and `d` are reduced and the transpose is applied to get output layout `ca`.
+
+**NOTE**: *Einsum* operation can perform on the single operand and in this case the operation can transpose the input and reduce its dimensions.
+
+**NOTE**: Input ranks must be equal to a length of corresponding subscripts and dimensions to which the same labels in input subscripts 
+correspond must be equal in size.
+
+**NOTE**: A labels can repeat in the same input subscript, for example, `equation` equal to `aac,abd,ddde`. In this case the corresponding dimensions
+must match in size and the operand the operand will be replaced by its diagonal along these dimensions.
+For example, *Einsum* operation on the single 3D tensor of shape `[2, 4, 5, 4]` with `equation` equal to `ijkj->ij`.
+
+**NOTE**: The specification considers the primitive algorithm for *Einsum* operation for better understanding of the operation
```

Comment:
Not sure if this Note is really needed. This is a common-sense that specification is not an implementation recomendation.

Reply:
I think extra reminder will not disturb the specification.